hs_tools_resource/utils.py

```diff
@@ -58,3 +62,48 @@ def get_SupportedSharingStatus_choices():
             ['Discoverable', 'Discoverable'],
             ['Private', 'Private'],
             ]
+
+
+def do_work_when_launching_app_as_needed(app_shortkey, res_shortkey, user):
```

Comment:
This method name sounds generic, but the method actually has a specific function of copying files to a federated irods.

Reply:
This method name is made generic on purpose since it is intended to handle other potential work needed to be done when a web app is being launched without being constrained to this special copying files to a federated irods server use case. The general docstring comment for this method should make this clear I believe.

Reply:
ok, that's fine, but I think we should probably keep the "work" separate from each other.  How do you see someone coming in and adding more pre-launch app code?  Are they to just place it inside this method with your code for federated irods copying?  I could see this method becoming large and ambiguous quickly.

I suggest either making this method name specific to your function or putting your irods copying work into it's own function and calling it here.